test/configure-provider.test.js

```diff
@@ -0,0 +1,64 @@
+var loopback = require('loopback');
+var m = require('./init');
+var mock = require('mock-require');
+var PassportConfigurator = require('../lib/passport-configurator.js');
+var assert = require('assert');
+
+describe('configureProvider', function() {
+  var app;
+  var passportConfigurator;
+
+  function LDAPStrategy(options, verify) {};
+  function LocalStrategy(options, verify) {};
+  function OAuth1Strategy(options, verify) {};
+  function OAuth2Strategy(options, verify) {};
+  function OpenIDStrategy(options, verify) {};
+  function OpenIDConnectStrategy(options, verify) {};
```

Comment:
Put all these inside "mock specific strategies" since they are only related to that before block.